### PR TITLE
feat: add fetch customers list

### PR DESCRIPTION
**Pull request by EmmanuelOmoiya**

### CODE REVIEW

**Review by NwokoyeChigozie — APPROVED** [review]

File: internal/models/user.go [hunk 1]
```diff
@@ -27,6 +29,7 @@ type CreateUserRequestModel struct {
 	LastName    string `json:"last_name" validate:"required"`
 	UserName    string `json:"username" validate:"required"`
 	PhoneNumber string `json:"phone_number"`
+	Role        string `json:"role"`
```
Comment (Cyberguru1) [inline internal/models/user.go:32]: is this okay?
a user can specify their role while registering?

File: pkg/repository/storage/postgresql/connection.go [hunk 2]
```diff
@@ -44,7 +44,7 @@ func connectToDb(host, user, password, dbname, port, sslmode, timezone string, l
 		port = detectedPort
 	}
 
-	dsn := fmt.Sprintf("host=%v user=%v password=%v dbname=%v port=%v sslmode=%v TimeZone=%v", host, user, password, dbname, port, sslmode, timezone)
+	dsn := fmt.Sprintf("host=%v user=%v password=%v dbname=%v port=%v sslmode=%v TimeZone=%v prefer_simple_protocol=true", host, user, password, dbname, port, sslmode, timezone)
```
Comment (Cyberguru1) [inline pkg/repository/storage/postgresql/connection.go:47]: is there a reason for  adding `prefer_simple_protocol=true`
it's causing a problem

Comment (EmmanuelOmoiya) [reply pkg/repository/storage/postgresql/connection.go:47]: Yeah, it was caching my statements, but you can remove it if it an error persists

File: internal/models/user.go (hunk 1 above)
Comment (EmmanuelOmoiya) [reply internal/models/user.go:32]: What do you suggest?

File: pkg/repository/storage/postgresql/connection.go (hunk 2 above)
Comment (vicradon) [reply pkg/repository/storage/postgresql/connection.go:47]: This thing you added broke our test pipeline bro

Comment (EmmanuelOmoiya) [reply pkg/repository/storage/postgresql/connection.go:47]: Apologies for the inconvinience

**Pull request merged**